bitcoin/src/crypto/taproot.rs

```diff
@@ -54,6 +58,24 @@ impl Signature {
         }
         ser_sig
     }
+
+    /// Serializes the signature (without heap allocation)
+    ///
+    /// This returns a type with an API very similar to that of `Box<[u8]>`.
+    /// You can get slice from it using deref coercions or turn it into iterator.
+    pub fn serialize(self) -> SerializedSignature {
+        let mut buf = [0; serialized_signature::MAX_LEN];
+        let ser_sig = self.sig.serialize();
+        buf[..64].copy_from_slice(&ser_sig);
+        let len = if self.hash_ty == TapSighashType::Default {
```

Comment:
Nit, why not just name this `hash_type`.

Reply:
IDK, I didn't come up with the name.

Reply:
I bet @apoelstra did it, he is one of those annoying C developers. Although we should totally start using `nbytes` as a variable name in rust `n_bytes` looks stupid.

(I purposely left of emojis because I believe this comment is funny if taken as a joke and even funnier if mis-taken to be serious.)

Reply:
In C you are allowed to use the word `type` as a keyword. In Rust you are not.

I could use `r#` identifiers, but given that I don't use trigraphs in C, I'm maybe not the right sort of greybeard :).

Reply:
`hash_type` is valid though.

Reply:
Yes, but using `type` when it appears as part of larger identifiers means it will be inconsistent with `type` when it appears by itself.

Reply:
Having the word "type" in sighash type is annoying also because its hard to write docs, "the foo type does ...", "the sighash type type does ..." - maybe we should use "sighash byte" always and stop calling it a type?

Reply:
The other common word in use is "flag". +1 to changing "type" to "flag" here.

Byte IMHO is misleading because it's secretly 4 bytes inside the sighash.